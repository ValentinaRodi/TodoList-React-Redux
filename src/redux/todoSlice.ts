import { createSlice, PayloadAction } from "@reduxjs/toolkit";
import { Todo } from "../models/Todo";
import { v4 as uuidv4 } from "uuid";

const initialState = [] as Todo[];

const todoSlice = createSlice({
  name: "todos",
  initialState,
  reducers: {
    addTodo: {
      reducer: (state: any[], action: PayloadAction<Todo>) => {
        state.push(action.payload);
      },
      prepare: (description: string) => ({
        payload: {
          id: uuidv4(),
          description,
          styleBtn: "item__btn",
          styleText: "item__text",
          counter: 1,
          itemShow: 2,
        } as Todo,
      }),
    },
    removeTodo(state: any[], action: PayloadAction<string>) {
      const index = state.findIndex((todo) => todo.id === action.payload);
      state.splice(index, 1);
    },
    setTodoStatus(
      state: any[],
      action: PayloadAction<{ id: string }>
    ) {
      const index = state.findIndex((todo) => todo.id === action.payload.id);
      state[index].styleBtn = "item__btn_check";
      state[index].styleText = "item__text_check";
      state[index].counter = 0;
    },
    showItemActive(state: any[], action: PayloadAction<{ todoItemShow: number }>) {
      state.map((item)=>item.itemShow = action.payload.todoItemShow)
    },
    removeTodoAll() {
      return initialState
    },
    
  },
});

export const { addTodo, removeTodo, setTodoStatus, showItemActive, removeTodoAll } = todoSlice.actions;
export default todoSlice.reducer;


