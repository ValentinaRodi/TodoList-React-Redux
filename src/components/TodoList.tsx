import { useState } from "react";
import  './todoList.css'

// Other Imports
import { useDispatch } from "react-redux";
import { AppDispatch } from "../redux/store";
import { addTodo } from "../redux/todoSlice";

export default function TodoList() {
   
    //React Hooks
    const [todoDescription, setTodoDescription] = useState("");

    //React Redux Hooks
    const dispatch = useDispatch<AppDispatch>();
    const dataTestId = "test-id";
    return (
        <div>
            <h1 className='title'>todos</h1>
            <div className='addItem'>
                <input
                    className='addItem__input'
                    placeholder='What needs to be done?'
                    onChange={(e) => setTodoDescription(e.target.value)}
                    value={todoDescription}
                    data-testid={dataTestId}
                />
                <button
                    className='addItem__btn'
                    onClick={() => {
                    dispatch(addTodo(todoDescription));
                    setTodoDescription("");
                    }}
                >âž•
                </button>
            </div>
        </div>
    )
}