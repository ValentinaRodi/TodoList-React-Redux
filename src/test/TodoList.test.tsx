import  TodoList  from '../components/TodoList';
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "../redux/store";

describe("<TodoList />", () => {
  it("should render successfully and match snapshot", () => {
    const { container } = render(
        <Provider store={store}> 
            <TodoList />
        </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});

