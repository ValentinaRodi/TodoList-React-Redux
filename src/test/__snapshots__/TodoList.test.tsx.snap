// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> should render successfully and match snapshot 1`] = `
<div>
  <div>
    <h1
      class="title"
    >
      todos
    </h1>
    <div
      class="addItem"
    >
      <input
        class="addItem__input"
        data-testid="test-id"
        placeholder="What needs to be done?"
        value=""
      />
      <button
        class="addItem__btn"
      >
        âž•
      </button>
    </div>
  </div>
</div>
`;
